egs/heroico/s5/run.sh

```diff
@@ -0,0 +1,451 @@
+	    data/lang/oovs_3g.txt \
+	    > \
+	    data/lang/lm_3g_no_oovs.txt
+
+    # replace epsilon symbol with \#0
+    utils/eps2disambig.pl \
+	< \
+	data/lang/lm_3g_no_oovs.txt \
+	| \
+	utils/s2eps.pl \
+	    > \
+	    data/lang/lm_3g_with_disambig_symbols_without_s.txt
+
+    # binarize the fst
+    fstcompile \
+	--isymbols=data/lang/words.txt \
+	--osymbols=data/lang/words.txt \
+	--keep_isymbols=false \
+	--keep_osymbols=false \
+	data/lang/lm_3g_with_disambig_symbols_without_s.txt \
+	data/lang/lm_3g_with_disambig_symbols_without_s.fst
+
+    # make the G.fst
+    fstarcsort \
+	data/lang/lm_3g_with_disambig_symbols_without_s.fst \
+	data/lang/G.fst
+
+    gzip \
+	language_models/lm_threegram.arpa
+fi
+
+if [ $stage -le 4 ]; then
+    # extract acoustic features
+    mkdir -p exp
+
+    for fld in native nonnative test train; do
+	if [ -e data/$fld/cmvn.scp ]; then
+	    rm data/$fld/cmvn.scp
+	fi
+
+	steps/make_mfcc.sh \
+	    --cmd "$train_cmd" \
+	    --nj 4 \
+	    data/$fld \
+	    exp/make_mfcc/$fld \
+	    mfcc || exit 1;
+
+	utils/fix_data_dir.sh \
+	    data/$fld || exit 1;
+
+	steps/compute_cmvn_stats.sh \
+	    data/$fld \
+	    exp/make_mfcc\
+	    mfcc || exit 1;
+
+	utils/fix_data_dir.sh \
+	    data/$fld || exit 1;
+	done
+fi
+
+if [ $stage -le 5 ]; then
+        echo "monophone training"
+    steps/train_mono.sh \
+	--nj 4 \
+	--cmd "$train_cmd" \
+	data/train \
+	data/lang \
+	exp/mono || exit 1;
+
+    # align with monophones
+    steps/align_si.sh \
+	--nj 8 \
+	--cmd "$train_cmd" \
+	data/train \
+	data/lang \
+	exp/mono \
+	exp/mono_ali || exit 1;
+fi
+
+if [ $stage -le 6 ]; then
+    echo "Starting  triphone training in exp/tri1"
+    steps/train_deltas.sh \
+	--cmd "$train_cmd" \
+	--cluster-thresh 100 \
+	1500 \
+	25000 \
+	data/train \
+	data/lang \
+	exp/mono_ali \
+	exp/tri1 || exit 1;
+
+    # align with triphones
+    steps/align_si.sh \
+	--nj 8 \
+	--cmd "$train_cmd" \
+	data/train \
+	data/lang \
+	exp/tri1 \
+	exp/tri1_ali
+fi
+
+if [ $stage -le 7 ]; then
+    echo "Starting (lda_mllt) triphone training in exp/tri2b"
+    steps/train_lda_mllt.sh \
+	--splice-opts "--left-context=3 --right-context=3" \
+	2000 \
+	30000 \
+	data/train \
+	data/lang \
+	exp/tri1_ali \
+	exp/tri2b
+
+    # align with lda and mllt adapted triphones
+    steps/align_si.sh \
+	--use-graphs true \
+	--nj 8 \
+	--cmd "$train_cmd" \
+	data/train \
+	data/lang \
+	exp/tri2b \
+	exp/tri2b_ali
+
+    echo "Starting (SAT) triphone training in exp/tri3b"
+    steps/train_sat.sh \
+	--cmd "$train_cmd" \
+	3100 \
+	50000 \
+	data/train \
+	data/lang \
+	exp/tri2b_ali \
+	exp/tri3b
+
+    # align with tri3b models
+    echo "Starting exp/tri3b_ali"
+    steps/align_fmllr.sh \
+	--nj 8 \
+	--cmd "$train_cmd" \
+	data/train \
+	data/lang \
+	exp/tri3b \
+	exp/tri3b_ali
+fi
+
+if [ $stage -le 8 ]; then
+    # i-vector  extractor training
+    local/nnet3/run_ivector_common.sh
+
+    # train and test chain models
+    local/chain/run_tdnn.sh
```

Comment:
the sequence of the two commands will y probably cause the run_ivector_common.sh to run twice. you can just call  
```
 local/chain/run_tdnn.sh
```

or other option that might be more "educative" is to run it like this
```
local/nnet3/run_ivector_common.sh
# we need to call the run_tdnn.sh with the stage param so that 
# the ivector script won't get run twice.
local/chain/run_tdnn.sh --stage 10
```